packages/shared/src/utils/os/os_detect.ts

```diff
@@ -89,3 +105,33 @@ export const mobileOSDetect = () => {
 
     return 'unknown';
 };
+
+export const mobileOSDetectAsync = async () => {
+    const userAgent = navigator.userAgent || navigator.vendor || window.opera || '';
+    const huaweiDevicesRegex =
+        /\b(ALP-|AMN-|ANA-|ANE-|ANG-|AQM-|ARS-|ART-|ATU-|BAC-|BLA-|BRQ-|CAG-|CAM-|CAN-|CAZ-|CDL-|CDY-|CLT-|CRO-|CUN-|DIG-|DRA-|DUA-|DUB-|DVC-|ELE-|ELS-|EML-|EVA-|EVR-|FIG-|FLA-|FRL-|GLK-|HMA-|HW-|HWI-|INE-|JAT-|JEF-|JER-|JKM-|JNY-|JSC-|LDN-|LIO-|LON-|LUA-|LYA-|LYO-|MAR-|MED-|MHA-|MLA-|MRD-|MYA-|NCE-|NEO-|NOH-|NOP-|OCE-|PAR-|PIC-|POT-|PPA-|PRA-|RNE-|SEA-|SLA-|SNE-|SPN-|STK-|TAH-|TAS-|TET-|TRT-|VCE-|VIE-|VKY-|VNS-|VOG-|VTR-|WAS-|WKG-|WLZ-|JAD-|MLD-|RTE-|NAM-|NEN-|BAL-|JLN-|YAL|MGA-|FGD-|XYAO-|BON-|ALN-|ALT-|BRA-|DBY2-|STG-|MAO-|LEM-|GOA-|FOA-|MNA-|LNA-)\b/i;
+
+    // Windows Phone must come first because its UA also contains "Android"
+    if (/windows phone/i.test(userAgent)) {
+        return 'Windows Phone';
+    }
+
+    if (/android/i.test(userAgent)) {
+        // Check if navigator.userAgentData is available for modern browsers
+        if (navigator?.userAgentData) {
```

Comment:
```suggestion
        if (navigator?.userAgentData || huaweiDevicesRegex.test(userAgent) || /huawei/i.test(userAgent)) {
```

Reply:
`userAgentData` doesn't have enough browser support. So, other two conditions are used as fallback of `userAgentData`

<img width="1410" alt="Screenshot 2024-04-12 at 12 06 32 PM" src="https://github.com/binary-com/deriv-app/assets/126637868/6b432940-881f-4cc4-96f6-ed323b797a39">